src/sentry/sentry_apps/models/sentry_app.py

```diff
@@ -220,7 +220,7 @@ def outboxes_for_delete(self) -> list[ControlOutbox]:
             ControlOutbox(
                 shard_scope=OutboxScope.APP_SCOPE,
                 shard_identifier=self.id,
-                object_identifier=self.id,
+                object_identifier=self.slug,
```

Comment:
I'm not sure if there is a better way to do this if I need id and slug in the receiver. I could imagine us queueing up an outbox with `object_identifier=self.id` but by the time it's processed, the code has been updated so the receiver expects `object_identifier` to be str and it could error out(?)